File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/ThuongHieuForm.ts
import { BooleanEditor, DateEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface ThuongHieuForm {
    TenThuongHieu: StringEditor;
    Slug: StringEditor;
    Logo: StringEditor;
    MoTa: StringEditor;
    ThuTu: IntegerEditor;
    HienThi: BooleanEditor;
    NoiBat: BooleanEditor;
    CreatedAt: DateEditor;
    UpdatedAt: DateEditor;
}

export class ThuongHieuForm extends PrefixedContext {
    static readonly formKey = 'SanPhamDB.ThuongHieu';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ThuongHieuForm.init) {
            ThuongHieuForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = BooleanEditor;
            var w3 = DateEditor;

            initFormType(ThuongHieuForm, [
                'TenThuongHieu', w0,
                'Slug', w0,
                'Logo', w0,
                'MoTa', w0,
                'ThuTu', w1,
                'HienThi', w2,
                'NoiBat', w2,
                'CreatedAt', w3,
                'UpdatedAt', w3
            ]);
        }
    }
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/ThuongHieu/ThuongHieuForm.cs
namespace wheystore.SanPhamDB.Forms;

[FormScript("SanPhamDB.ThuongHieu")]
[BasedOnRow(typeof(ThuongHieuRow), CheckNames = true)]
public class ThuongHieuForm
{
    public string TenThuongHieu { get; set; }
    public string Slug { get; set; }
    public string Logo { get; set; }
    public string MoTa { get; set; }
    public int ThuTu { get; set; }
    public bool HienThi { get; set; }
    public bool NoiBat { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}